src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/EggFinder.java

```diff
@@ -0,0 +1,146 @@
+package de.hysky.skyblocker.skyblock.chocolatefactory;
+
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.utils.ColorUtils;
+import de.hysky.skyblocker.utils.Constants;
+import de.hysky.skyblocker.utils.waypoint.Waypoint;
+import it.unimi.dsi.fastutil.objects.ObjectImmutableList;
+import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.component.DataComponentTypes;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.decoration.ArmorStandEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.text.Text;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static de.hysky.skyblocker.skyblock.chocolatefactory.EggFinder.EggType.BREAKFAST;
+import static de.hysky.skyblocker.skyblock.chocolatefactory.EggFinder.EggType.DINNER;
+import static de.hysky.skyblocker.skyblock.chocolatefactory.EggFinder.EggType.LUNCH;
+
+public class EggFinder {
+	private static final Pattern eggFoundPattern = Pattern.compile("^(?:HOPPITY'S HUNT You found a Chocolate|You have already collected this Chocolate) (Breakfast|Lunch|Dinner)");
+	private static final Pattern newEggPattern = Pattern.compile("^HOPPITY'S HUNT A Chocolate (Breakfast|Lunch|Dinner) Egg has appeared!$");
+	private static final Logger logger = LoggerFactory.getLogger("Skyblocker Egg Finder");
+
+	private EggFinder() {
+	}
+
+	public static void init() {
+		ClientPlayConnectionEvents.JOIN.register((ignored, ignored2, ignored3) -> clearEggs());
+		ClientReceiveMessageEvents.GAME.register(EggFinder::onChatMessage);
+		WorldRenderEvents.AFTER_TRANSLUCENT.register(EggFinder::renderWaypoints);
+	}
+
+	public static void checkIfEgg(Entity entity) {
+		if (!SkyblockerConfigManager.get().helpers.chocolateFactory.enableEggFinder) return;
+		if (BREAKFAST.egg.getValue() != null && DINNER.egg.getValue() != null && LUNCH.egg.getValue() != null) return; //Don't check for eggs if we already found all of them
+		if (!(entity instanceof ArmorStandEntity armorStand) || armorStand.hasCustomName() || !armorStand.isInvisible() || !armorStand.shouldHideBasePlate()) return;
+		for (ItemStack itemStack : armorStand.getArmorItems()) {
+			try {
+				if (!itemStack.isEmpty() || itemStack.getItem() == Items.PLAYER_HEAD) {
+					String texture = itemStack.getComponents()
```

Comment:
ItemUtils has a method for this, perhaps while you're at it you could change it to use your way of iterating which should be faster than streams.